core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -2285,8 +2285,9 @@ private static SqlNode createSqlDataTypeSpecByName(String typeAlias, RelDataType
         if (isContainsPrecision) {
           String dataType = getDataTypeBasedOnPrecision(precision, scale);
           if (!isContainsNegativePrecisionOrScale) {
-            typeAlias = precision > 0 ? isContainsScale ? dataType + "(" + precision + ","
-                + scale + ")" : dataType + "(" + precision + ")" : dataType;
+            typeAlias = precision > 0 && !(scale >= 127) ? isContainsScale
```

Comment:
you can only have scale upto 38 even with BIGNUMERIC.
https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types